transport/src/main/java/io/netty/channel/SimpleChannelEventHandler.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2013 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.internal.TypeParameterMatcher;
+
+/**
+ * {@link ChannelInboundHandlerAdapter} which allows to conveniently only handle a specific type of user events.
+ *
+ * For example, here is an implementation which only handle {@link String} user events.
+ *
+ * <pre>
+ *     public class StringEventHandler extends
+ *             {@link SimpleChannelEventHandler}&lt;{@link String}&gt; {
+ *
+ *         {@code @Override}
+ *         protected void eventReceived({@link ChannelHandlerContext} ctx, {@link String} evt)
+ *                 throws {@link Exception} {
+ *             System.out.println(evt);
+ *         }
+ *     }
+ * </pre>
+ */
+public abstract class SimpleChannelEventHandler<I> extends ChannelInboundHandlerAdapter {
+
+    private final TypeParameterMatcher matcher;
+
+    /**
+     * Create a new instance which will try to detect the types to match out of the type parameter of the class.
+     */
+    protected SimpleChannelEventHandler() {
+        matcher = TypeParameterMatcher.find(this, SimpleChannelEventHandler.class, "I");
+    }
+
+    /**
+     * Create a new instance
+     *
+     * @param userEventType The type of user events to match
+     */
+    protected SimpleChannelEventHandler(Class<? extends I> userEventType) {
+        matcher = TypeParameterMatcher.get(userEventType);
+    }
+
+    /**
+     * Returns {@code true} if the given user event should be handled. If {@code false} it will be passed to the next
+     * {@link ChannelInboundHandler} in the {@link ChannelPipeline}.
+     */
+    public boolean acceptEvent(Object evt) throws Exception {
+        return matcher.match(evt);
+    }
+
+    @Override
+    public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
+        if (acceptEvent(evt)) {
+            @SuppressWarnings("unchecked")
+            I ievt = (I) evt;
+            eventReceived(ctx, ievt);
```

Comment:
I think we should follow the same pattern as in `SimpleChannelInboundHandler` and allow to automatically call `release` on the event.

Reply:
I wasn't sure how often `release` is applicable to events, but I agree it probably makes more sense, and it's more consistent with the "Simple" naming convention.  Changing.